DataStructures/MinMaxHeap.cs

```diff
@@ -0,0 +1,373 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+
+namespace DataStructures
+{
+    /// <summary>
+    /// This class implements min-max heap.
+    /// It provides functionality of both min-heap and max-heap with the same time complexity.
+    /// Therefore it provides constant time retrieval and logarithmic time removal
+    /// of both the minimum and maximum elements in it.
+    /// </summary>
+    /// <typeparam name="T">Generic type.</typeparam>
+    public class MinMaxHeap<T>
+    {
+        private readonly List<T> heap = new List<T>();
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Remove the member initializer, all constructors set an initial value for the member.](https://app.codacy.com/manual/siriak/C-Sharp/pullRequest?prid=5421646)